src/site/content/en/blog/service-worker-mindset/index.md

```diff
@@ -0,0 +1,204 @@
+---
+title: Service Worker Mindset
+subhead: How to think when thinking about Service Workers.
+authors:
+  - geddski
+date: 2019-05-28
+hero: hero.jpg
+hero_position: center
+alt: Service Workers lining up.
+description: |
+  Working with Service Workers is new and unfamiliar for many web devs. Here's some tips for wrapping your mind around them.
+tags:
+  - post # post is a required tag for the article to show up in the blog.
+  - service-worker
+  - PWA
+  - offline
+  - caching
+---
+
+Service Workers are powerful and absolutely worth mastering. They'll enable you to deliver an entirely new level of experience to your users. But Service Workers are unlike anything most of us web devs are used to. They come with a steep learning curve and a handful of snags you've got to watch out for.
+
+The best way to ensure success with Service Workers is to have the right mindset when working with them. In this post we'll wrap our brains around the paradoxical traits that make Service Workers both tricky and awesome.
+
+## The Same, but Different
+
+While coding your Service Worker, many things will feel familiar. You get to use your favorite new JavaScript language features. You listen to lifecycle events just like with UI events. You manage control flow with promises like you're used to.
+
+But other Service Worker behavior causes you to scratch your head in confusion. Especially when you refresh the page and don't see your code changes applied.
+
+### A New Layer
+
+Normally when building a site you have just two layers to think about: the client and the server. Service Worker is a brand new layer that sits in the middle.
+
+![Middle Layer](./middle-layer.jpg)
```

Comment:
I think you can expand on the alt text a bit and try to describe the scene to a non-sighted user.

Except in cases where you think the image is purely decorative. In those instances you can leave the alt text blank:

```
![](./some-image.jpg)
```

Reply:
ahh indeed. improved them.